### PR TITLE
Use stochastic restart patterns on rerun

### LINKED ISSUE #2937: stage atm_stoch.res.nc and ocn_stoch.res.nc during segment forecasts
### What new functionality do you need?

When running ensembles in segments (i.e., restarting the model), the stoch restart files are needed to start the model with the same stochastic state. 

### What are the requirements for the new functionality?

atm_stoch.res.nc and ocn_stoch.res.nc files need to be used in a re-run or segment ensemble forecast 

### Acceptance Criteria

atm_stoch.res.nc and ocn_stoch.res.nc files are used in segment forecasts and forecasts that re-start

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
   
The stochastic pattern restart files were not being copied into the input directory when restarting the model after a segment/failure. These files are now copied in and the stochini flag set to `.true.` in the namelist on a rerun.
    
The files are NOT copied in for non-rerun warm starts.

Also removes the restriction that stochastic physics cannot be run on member 0, as this is desired down the line. Additional settings are added to the fcst and efcs configs to retain the current behavior.

A bug was also discovered and corrected during this work. The stage, forecast, and archive job all assumed that `ca_data` tile files are always present, but these are only created when cellular automata is on. Now `ca_data` files are handled if CA is on. To implement this change, the `DO_CA` setting had to be moved from the forecast configs to base so it is available to the stage_ic and archive jobs.

Resolves #2937

# Type of change
- [x] Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO

# How has this been tested?
Multi-segment test on Hercules

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

**Review by aerorahul — DISMISSED** [review]
looks good to me.  No comments.
Please invite review from @pjpegion @NeilBarton-NOAA

File: ush/forecast_postdet.sh [hunk 1]
```diff
@@ -47,7 +47,19 @@ FV3_postdet() {
       || ( echo "FATAL ERROR: Unable to copy FV3 IC, ABORT!"; exit 1 )
     done
 
-    if [[ "${RERUN}" != "YES" ]]; then
+    if [[ "${RERUN}" == "YES" ]]; then
```
Comment (NeilBarton-NOAA) [inline ush/forecast_postdet.sh:50]: @bingfu-NOAA @EricSinsky-NOAA Is there a time that we want to start the model with these files? i.e., use them in more ways than a RERUN?

Comment (bingfu-NOAA) [reply ush/forecast_postdet.sh:50]: For 00Z, we will stop at 16-day forecast, then warm start the model to run 16-48 days forecast segment. We will need those files at that time besides a RERUN.

File: parm/config/gefs/config.fcst [hunk 2]
```diff
@@ -227,11 +227,13 @@ case ${imp_physics} in
...
-export DO_SPPT=${DO_SPPT:-"NO"}
-export DO_SKEB=${DO_SKEB:-"NO"}
-export DO_SHUM=${DO_SHUM:-"NO"}
-export DO_LAND_PERT=${DO_LAND_PERT:-"NO"}
-export DO_CA=${DO_CA:-"YES"}
+export DO_SPPT="NO"
+export DO_SKEB="NO"
+export DO_SHUM="NO"
+export DO_LAND_PERT="NO"
+export DO_CA="NO"
```
Comment (bingfu-NOAA) [inline parm/config/gefs/config.fcst]: is this config file being used by mem000 in ensemble?, if so, then DO_CA should be YES.

File: parm/config/gfs/config.fcst [hunk 3]
```diff
@@ -257,14 +257,13 @@ case ${imp_physics} in
...
-export DO_LAND_PERT=${DO_LAND_PERT:-"NO"}
-export DO_CA=${DO_CA:-"YES"}
-if [[ "${DO_NEST:-NO}" == "YES" ]] ; then
-  export DO_CA="NO"  # CA does not work with nesting.
-fi
+export DO_SPPT="NO"
+export DO_SKEB="NO"
+export DO_SHUM="NO"
+export DO_LAND_PERT="NO"
+export DO_CA="NO"
```
Comment (bingfu-NOAA) [inline parm/config/gfs/config.fcst]: GFS is using CA, so DO_CA should be YES.

File: ush/forecast_postdet.sh (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:50]: Subsequent segments count as RERUNs inside the system.

File: parm/config/gefs/config.fcst (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/gefs/config.fcst]: Done

File: parm/config/gfs/config.fcst (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/gfs/config.fcst]: Done

File: parm/config/gfs/config.base [hunk 4]
```diff
@@ -448,6 +448,15 @@ export INCVARS_EFOLD="5"
 export netcdf_diag=".true."
 export binary_diag=".false."
 
+# Cellular automata
+case ${RUN} in
```
Comment (CatherineThomas-NOAA) [inline parm/config/gfs/config.base]: I think this is leftover from mixed `DO_CA` settings, but is the `case` section necessary now?

**Comment by WalterKolczynski-NOAA** [comment]
This works for me manually. I'm going to try again and inspect it as it runs.

File: parm/stage/atmosphere_warm.yaml.j2 [hunk 5]
```diff
@@ -15,9 +15,12 @@ atmosphere_warm:
         {% for ftype in ["coupler.res", "fv_core.res.nc"] %}
         - ["{{ ICSDIR }}/{{ COMOUT_ATMOS_RESTART_PREV_MEM | relpath(ROTDIR) }}/{{ m_prefix }}.{{ ftype }}", "{{ COMOUT_ATMOS_RESTART_PREV_MEM }}"]
         {% endfor %}
-        {% for ftype in ["ca_data", "fv_core.res", "fv_srf_wnd.res", "fv_tracer.res", "phy_data", "sfc_data"] %}
```
Comment (bingfu-NOAA) [inline parm/stage/atmosphere_warm.yaml.j2:18]: What is the reason to remove ca_data?

File: ush/forecast_predet.sh [hunk 6]
```diff
@@ -377,41 +396,38 @@ FV3_predet(){
...
-      ISEED_LNDP=$(( (base_seed + 5) % 2147483647 ))
-      LNDP_TAU=${LNDP_TAU:-21600}
-      LNDP_SCALE=${LNDP_SCALE:-500000}
-      lndp_var_list=${lndp_var_list:-"'smc', 'vgf',"}
-      lndp_prt_list=${lndp_prt_list:-"0.2,0.1"}
-      n_var_lndp=$(echo "${lndp_var_list}" | wc -w)
-    fi
+  if [[ "${DO_CA:-}" == "YES" ]]; then
+    do_ca=".true."
+    ISEED_CA=$(( (base_seed + 18) % 2147483647 ))
```
Comment (bingfu-NOAA) [inline ush/forecast_predet.sh:419]: As CA is being used by both GFS/GEFS, probably we can just use
ISEED_CA=$(( (base_seed) % 2147483647 ))

File: parm/stage/atmosphere_warm.yaml.j2 (hunk 5 above)
Comment (WalterKolczynski-NOAA) [reply parm/stage/atmosphere_warm.yaml.j2:18]: It's not removed, it is just only added to the list when CA is on.

File: parm/config/gefs/config.base [hunk 7]
```diff
@@ -287,10 +287,17 @@ export DO_MERGENSST="NO"
...
-# this will be overwritten for the perturbed members
+# this will be overwridden for the perturbed members
 export ENSMEM=${ENSMEM:-"000"}
 export MEMDIR="mem${ENSMEM}"
 
+# cellular automata
+if (( ENSMEM == 0 )); then
+    export DO_CA="YES"
+else
+    export DO_CA="YES"
```
Comment (DavidHuber-NOAA) [inline parm/config/gefs/config.base:298]: Should one of these be `NO`?

Comment (WalterKolczynski-NOAA) [reply parm/config/gefs/config.base:298]: Nope, at least not by default right now.

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
CI Tests set up to run in /lfs/h2/emc/ptmp/walter.kolczynski/PR/PR_3077/RUNTESTS on WCOSS

**Comment by WalterKolczynski-NOAA** [comment]
Got all the way to the last cycle of the extended test and then died. Investigating.

**Comment by WalterKolczynski-NOAA** [comment]
Looks like stmp filled on WCOSS

**Review by KateFriedman-NOAA — APPROVED** [review]
I can't comment on the science updates in this PR but the workflow changes look good. Thanks @WalterKolczynski-NOAA !

**Pull request merged**